pkg/pdatatest/ptracetest/traces.go

```diff
@@ -301,3 +297,84 @@ func CompareSpan(expected, actual ptrace.Span) error {
 
 	return errs
 }
+
+// CompareEvents compares each part of two given SpanEventSlice and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareEvents(expected, actual ptrace.SpanEventSlice) (errs error) {
```

Comment:
Please don't expose it, we don't expose compare functions for slices here

```suggestion
func compareSpanEventSlice(expected, actual ptrace.SpanEventSlice) (errs error) {
```